packages/core/src/index.ts

```diff
@@ -205,22 +205,21 @@ class Hopr extends EventEmitter {
 
     this.heartbeat = new Heartbeat(this.networkPeers, subscribe, sendMessageAndExpectResponse, hangup)
 
-    subscribeToAcknowledgements(subscribe, this.db, await this.paymentChannels, (ack) =>
-      this.emit('message-acknowledged:' + ack.getKey())
+    subscribeToAcknowledgements(subscribe, this.db, await this.paymentChannels, this.getId(), (ack) =>
+      this.emit('message-acknowledged:' + ack.ackChallenge)
     )
 
     const onMessage = (msg: Uint8Array) => this.emit('hopr:message', msg)
     this.forward = new PacketForwardInteraction(
-      this.db,
-      await this.paymentChannels,
-      this.getId(),
-      this.libp2p,
       subscribe,
       sendMessage,
-      onMessage
+      this.getId(),
+      await this.paymentChannels,
+      onMessage,
+      this.db
```

Comment:
why did you reorder this?

Reply:
not intentionally ... I've done the flattening to the forward logic and just in added the parameters one-by-one. I have no strong opinion here